PowerPointLabs/PowerPointLabs/AudioMisc/Audio.cs

```diff
@@ -41,6 +45,27 @@ public Audio(string name, string saveName, int matchScriptID)
             Type = AudioHelper.GetAudioType(saveName);
         }
 
+        public Audio(Shape shape, string saveName)
+        {
+            // detect audio type from shape tag
+            AudioType audioType;
+            if (!Enum.TryParse<AudioType>(shape.Tags[AudioTypeTagName], out audioType))

```

Comment:
When audio is recorded/generated, it creates an audio object, sets its type, then calls the function to add to slide.

So (wav file -> audio object -> shape) has no problem

The issue is that when we take a shape and create an audio object from it, it checks for the sampling rate to determine the type. So (shape -> audio object) is an issue. By setting a tag to indicate the type from audio object -> shape, we can get shape -> audio object without using the sampling rate.